### PR TITLE
PepStore_Qiuyang

### PR DESCRIPTION
**Pull request by FlorianQU**

### CODE REVIEW

File: PetApi/Controllers/PetStoreController.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        [HttpDelete("deletePet")]
+        public void DeletePet(string name)
+        {
+            pets = pets.Where(pet => pet.Name != name).ToList();
+        }
+
+        [HttpPut("updatePet")]
+        public Pet UpdatePetPrice(string name, Pet newpet)
+        {
+            pets = pets.Select(pet =>
```
Comment (YYuan1414) [inline PetApi/Controllers/PetStoreController.cs:67]: use pets.Single since the name is unique

File: PetApiTest/PetApiTest.cs [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+            // then
+            response.EnsureSuccessStatusCode();
+            var responseString = await response.Content.ReadAsStringAsync();
+            List<Pet> actualPet = JsonConvert.DeserializeObject<List<Pet>>(responseString);
+            Assert.Equal(new List<Pet>() { pet_1, pet_3 }, actualPet);
+        }
+
+        [Fact]
+        public async Task Should_Return_Correct_Pet_When_Given_Price_Range()
```
Comment (YYuan1414) [inline PetApiTest/PetApiTest.cs:224]: add the test that the expected is null

File: PetApi/Controllers/PetStoreController.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        [HttpDelete("clear")]
+        public void Clear()
+        {
+            pets.Clear();
+        }
+
+        [HttpDelete("deletePet")]
+        public void DeletePet(string name)
+        {
+            pets = pets.Where(pet => pet.Name != name).ToList();
```
Comment (kristol07) [inline PetApi/Controllers/PetStoreController.cs:61]: you didn't delete the pet in pets, but just return a filtered list of pets

File: PetApi/Controllers/PetStoreController.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        }
+
+        [HttpPut("updatePet")]
+        public Pet UpdatePetPrice(string name, Pet newpet)
+        {
+            pets = pets.Select(pet =>
+            {
+                if (pet.Name == name)
+                {
```
Comment (kristol07) [inline PetApi/Controllers/PetStoreController.cs:70]: update directly is better than return new
